### PR TITLE
[primus] Add missing method 'open' to Primus.Socket class

### PR DESCRIPTION
**Pull request by alaadotsol**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/primus/primus#primusopen
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/primus/index.d.ts [hunk 1]
```diff
@@ -192,5 +193,6 @@ declare namespace Primus {
         on(event: 'reconnect timeout' | 'reconnect failed', handler: (err: Error, opts: ReconnectEventOpts) => void): this;
         on(event: 'data', handler: (message: any) => void): this;
         on(event: 'error', handler: (err: Error) => void): this;
+    	open(): void;
```
Comment (danger-public) [inline types/primus/index.d.ts:196]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## primus ([<kbd>unpkg</kbd>](https://unpkg.com/browse/primus@latest/))
was missing the following properties:
1. transformers
1. parsers
1. Transformer
1. Spark
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a9bd1e6e72939f913baaa3ab1c9b45d65ba858b2
</p>

File: types/primus/index.d.ts [hunk 2]
```diff
@@ -192,5 +193,6 @@ declare namespace Primus {
         on(event: 'reconnect timeout' | 'reconnect failed', handler: (err: Error, opts: ReconnectEventOpts) => void): this;
         on(event: 'data', handler: (message: any) => void): this;
         on(event: 'error', handler: (err: Error) => void): this;
+        open(): void;
```
Comment (tellnes) [inline types/primus/index.d.ts:196]: @sheetalkamat IIRC correctly it returns `this` and not undefined.

Comment (AlaaZorkane) [reply types/primus/index.d.ts:196]: Yeah, you're right!

Comment (AlaaZorkane) [reply types/primus/index.d.ts:196]: #45063 I made a pull request fixing it, feel free to review it 🙂

**Pull request merged**